android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -102,29 +103,14 @@ class RemoteSettingsClient(
             try {
                 val fileReader = FileReader(file)
                 val jsonObject = JSONObject(fileReader.readText())
-                val jsonArray = jsonObject.getJSONArray("records")
-                val records = mutableListOf<RemoteSettingsRecord>()
-
-                for (i in 0 until jsonArray.length()) {
-                    val recordJson = jsonArray.getJSONObject(i)
-                    val id = recordJson.getString("id")
-                    val lastModified = recordJson.getString("lastModified").toULong()
-                    val deleted = recordJson.getBoolean("deleted")
-                    val attachment = recordJson.opt("attachment")
-                    val fields = recordJson.opt("fields")
-
-                    records.add(
-                        RemoteSettingsRecord(
-                            id = id,
-                            lastModified = lastModified,
-                            deleted = deleted,
-                            attachment = attachment as Attachment?,
-                            fields = fields as RsJsonObject,
-                        ),
-                    )
-                }
+                val records: Any? = jsonObject.get("records")
+
                 fileReader.close()
-                RemoteSettingsResult.Success(records)
+                if (records is RemoteSettingsResponse) {
+                    RemoteSettingsResult.Success(records)
+                } else {
```

Comment:
Bit of oddness here - we're `get`ting the "records" field but the expected type is `RemoteSettingsResponse`? In general, let's try to use `response` as a variable name where appropriate. If possible, I would also prefer if we could try to avoid the `Any?` type and instead directly cast it. Something like
```
val response = jsonObject.get("<name>") as? RemoteSettingsResponse
```